### PR TITLE
use timer to count down time

### PR DESCRIPTION
**Pull request by JieJhih**
#### Submitter checklist

- [ ] Change is code complete and matches issue description.
- [x] Change is covered by existing or new tests.

#### Description
Use timer to count down time.

#### Which issue this PR fixes

fixes #

### CODE REVIEW

File: pkg/gits/gitea.go [hunk 1]
```diff
@@ -175,17 +175,18 @@ func (p *GiteaProvider) ForkRepository(originalOrg string, name string, destinat
...
+			defer timer.Stop()
 			for {
-				time.Sleep(5 * time.Second)
-				repo, err = p.Client.GetRepo(owner, name)
-				if repo != nil && err == nil {
-					break
-				}
-				t := time.Now()
-				if t.After(deadline) {
+				select {
```
Comment (hferentschik) [inline pkg/gits/gitea.go]: Looking basically ok. Just wondering what your main goal is? Is there a particular issue you want to address or you just want to use more idiomatic Go?

Comment (JieJhih) [reply pkg/gits/gitea.go]: Yeah, it just cleanup.
Deprecated ``deadline`` and ``t`` variables, use one variable I think that makes the code cleaner.

File: pkg/gits/github.go [hunk 2]
```diff
@@ -325,17 +325,18 @@ func (p *GitHubProvider) ForkRepository(originalOrg string, name string, destina
...
-					break
-				}
-				t := time.Now()
-				if t.After(deadline) {
+				select {
+				case <-timer.C:
 					return nil, fmt.Errorf("Gave up waiting for Repository %s/%s to appear: %s", owner, name, err)
+				default:
+					time.Sleep(5 * time.Second)
+					repo, _, err = p.Client.Repositories.Get(p.Context, owner, name)
```
Comment (ccojocar) [inline pkg/gits/github.go]: We should definitely use `context.WithTimeout` instead to cancel the request when timeout expires.

https://golang.org/pkg/context/#WithTimeout

**Comment by hferentschik** [comment]
@JieJhih, any chance you want to rebase and update this PR? Otherwise, I think I am going to close it.

**Comment by hferentschik** [comment]
@@JieJhih, I am going to close this PR for now. If you feel for rebasing and applying the requested changes, just create a new PR. Thanks.

**Pull request closed without merging**